packages/instrumentation-langchain/test/instrumentation.test.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Traceloop
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from "assert";
+
+import { context } from "@opentelemetry/api";
+import { AsyncHooksContextManager } from "@opentelemetry/context-async-hooks";
+import {
+  BasicTracerProvider,
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from "@opentelemetry/sdk-trace-base";
+
+import type * as ToolsModule from "langchain/tools";
+import type * as AgentsModule from "langchain/agents";
+import type * as ChainsModule from "langchain/chains";
+import { ChatPromptTemplate, PromptTemplate } from "@langchain/core/prompts";
+import { createOpenAIToolsAgent } from "langchain/agents";
+import { Calculator } from "langchain/tools/calculator";
+import { pull } from "langchain/hub";
+import { HNSWLib } from "@langchain/community/vectorstores/hnswlib";
+import { ChatOpenAI, OpenAI, OpenAIEmbeddings } from "@langchain/openai";
+import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
+
+import { LangChainInstrumentation } from "../src/instrumentation";
+
+const memoryExporter = new InMemorySpanExporter();
+
+describe("Test LlamaIndex instrumentation", () => {
+  const provider = new BasicTracerProvider();
+  let instrumentation: LangChainInstrumentation;
+  let contextManager: AsyncHooksContextManager;
+  let langchainAgentsModule: typeof AgentsModule;
+  let langchainToolsModule: typeof ToolsModule;
+  let langchainChainsModule: typeof ChainsModule;
+
+  before(() => {
+    provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+    instrumentation = new LangChainInstrumentation();
+    instrumentation.setTracerProvider(provider);
+    langchainAgentsModule = require("langchain/agents");
+    langchainToolsModule = require("langchain/tools");
+    langchainChainsModule = require("langchain/chains");
+  });
+
+  beforeEach(() => {
+    contextManager = new AsyncHooksContextManager().enable();
+    context.setGlobalContextManager(contextManager);
+  });
+
+  afterEach(() => {
+    memoryExporter.reset();
+    context.disable();
+  });
+
+  it("should set attributes in span for tools and agent instrumentation", async () => {
+    const llm = new ChatOpenAI({});
+    const tools = [new Calculator(), new langchainToolsModule.SerpAPI()];
```

Comment:
Can we remove the dependency on the SerpAPI?

Reply:
Added test which use `WikipediaQueryRun` instead of `SerpAPI`.